private/dbutil/pgutil/db.go

```diff
@@ -92,8 +92,16 @@ func QuerySnapshot(ctx context.Context, db dbschema.Queryer) (*dbschema.Snapshot
 }
 
 // CheckApplicationName ensures that the Connection String contains an application name.
-func CheckApplicationName(s string) (r string) {
+func CheckApplicationName(s string, app string) (r string) {
 	if !strings.Contains(s, "application_name") {
+		if app != "" {
```

Comment:
Do you think we could get away with failing here if `app == ""`? That is to say, can we _require_ a non-empty application name?

Reply:
I like that idea actually.